src/3rd_party_config.c

```diff
@@ -0,0 +1,235 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/config_parser.h"
+#include "lib/list.h"
+#include "lib/sys.h"
+#include "swupd.h"
+#include <errno.h>
+#include <stdlib.h>
+
+#ifdef THIRDPARTY
+#define MAX_REPO_ALLOWED_FIELDS 2
+
+/** @brief Internal Memory representation of a repo.  */
+struct repo {
+	char *name;
+	char *url;
+};
+
+/** @brief List of repos. */
+struct list *repos_kv;
+
+/**
+ * @brief This function is called by the repo ini parser It is called
+
+ *
+ * @param section A string containing section of the file
+ * @param key A string containing key
+ * @param value A string containing value
+ *
+ * @returns True for every valid line which could be extracted, false if either
+ * section, key or value is NULL
+ */
+bool parse_key_values(char *section, char *key, char *value, void UNUSED_PARAM *data)
+{
+	static int repo_field_count = 0;
```

Comment:
you don't need to count the parsed values nor you have to use a static repo struct here, you can just store the "section" and "value" whenever "key" = "url" and ignore all other cases, that way you will have all the required data in the same function call. For reference look at the "load_repos" function in PR https://github.com/clearlinux/swupd-client/pull/1192/files#diff-1550ff8d89cd48ddcbe86c76f7f79f33

Reply:
Thanks for sharing. That is pretty smart! 👍

Reply:
Thanks will change in next push